src/plugins/usage_collection/server/collector/collector_set.ts

```diff
@@ -39,10 +39,22 @@ export class CollectorSet {
     this.maximumWaitTimeForAllCollectorsInS = maximumWaitTimeForAllCollectorsInS || 60;
   }
 
-  public makeStatsCollector = <T, U>(options: CollectorOptions<T, U>) => {
+  public makeStatsCollector = <
+    T,
+    U,
+    O extends CollectorOptions<T, U> = CollectorOptions<T, U> // Used to allow extra properties (the Collector constructor extends the class with the additional options provided)
```

Comment:
This is to allow `x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_settings_collector.ts` to send the additional method `getEmailValueStructure` without TS complaining about it.

The logic inside the Collector class actually allows doing that.

In a follow-up PR I'll update the types inside the `Collector` class to better define the `this` context in the `isReady`, `fetch` and `formatForBulkUpload` methods. It requires some mocks to be updated in tests (production logic is good as it is for now), and I don't want to make the scope of this PR explode 🙂